src/v/iceberg/rest_client/catalog_client.cc

```diff
@@ -224,7 +225,14 @@ ss::future<expected<iobuf>> catalog_client::perform_request(
         }
 
         retriable_errors.emplace_back(std::move(error.err));
-        co_await ss::sleep_abortable(permit.delay, rtc.root_abort_source());
+        auto sleep_fut = co_await ss::coroutine::as_future(
+          ss::sleep_abortable(permit.delay, rtc.root_abort_source()));
+        if (sleep_fut.failed()) {
```

Comment:
don't you need to call `get` on the non-failure path?

Reply:
Mm I don't think so? There's nothing in the result in the non-failure path because it's a future<void>. At least, it hasn't been complaining about an ignored future, if that's the concern, and this pattern does appear in at least a couple places in the codebase. Have you seen it be problematic?